### PR TITLE
MSONAR-184 Supported values with commas in multi-values properties

### PR DESCRIPTION
**Pull request by ADarko22**
Supported values with commas in multi-values properties by wrapping them with double quotes, according to CSV format

### CODE REVIEW

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenUtils.java [hunk 1]
```diff
@@ -89,4 +91,27 @@ static void putAll(Properties src, Map<String, String> dest) {
     }
   }
 
+  /**
+   * Joins a list of strings that may contain commas by wrapping those strings in double quotes, like in CSV format.
+   * <p>
+   * For example:
+   *  values = { "/home/users/me/artifact-123,456.jar", "/opt/lib" }
+   *  separator = ","
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenUtils.java:99]: We can probably remove this line now that the separator is constant
```suggestion
```

File: its/projects/shared/struts-1.3.9-diet/core/pom.xml [hunk 2]
```diff
@@ -25,7 +25,7 @@
     <parent>
         <groupId>org.apache.struts</groupId>
         <artifactId>struts-parent</artifactId>
-        <version>1.3.9</version>
+        <version>1,3,9</version>
```
Comment (henryju) [inline its/projects/shared/struts-1.3.9-diet/core/pom.xml:28]: Personally, I like to keep ITs looking like "real" projects. Here we are testing a corner case, that has no dependency on the SonarQube version, so I would go for an "invoker" test.

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java [hunk 3]
```diff
@@ -385,10 +383,10 @@ private void synchronizeFileSystemAndOtherProps(MavenProject pom, Map<String, St
     MavenUtils.putAll(userProperties, props);
 
     List<File> mainDirs = mainSources(pom);
-    props.put(ScanProperties.PROJECT_SOURCE_DIRS, StringUtils.join(toPaths(mainDirs), SEPARATOR));
+    props.put(ScanProperties.PROJECT_SOURCE_DIRS, MavenUtils.joinAsCsv(toPaths(mainDirs)));
     List<File> testDirs = testSources(pom);
     if (!testDirs.isEmpty()) {
-      props.put(ScanProperties.PROJECT_TEST_DIRS, StringUtils.join(toPaths(testDirs), SEPARATOR));
+      props.put(ScanProperties.PROJECT_TEST_DIRS,  MavenUtils.joinAsCsv(toPaths(testDirs)));
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java:389]: Let's remove unnecessary spaces
```suggestion
      props.put(ScanProperties.PROJECT_TEST_DIRS, MavenUtils.joinAsCsv(toPaths(testDirs)));
```

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java [hunk 4]
```diff
@@ -425,7 +423,7 @@ private static void populateLibraries(MavenProject pom, Map<String, String> prop
         .forEach(libraries::add);
     }
     if (!libraries.isEmpty()) {
-      String librariesValue = StringUtils.join(toPaths(libraries), SEPARATOR);
+      String librariesValue =  MavenUtils.joinAsCsv(toPaths(libraries));
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java:426]:
```suggestion
      String librariesValue = MavenUtils.joinAsCsv(toPaths(libraries));
```

File: its/projects/project-default-config/pom.xml [hunk 5]
```diff
@@ -21,8 +21,8 @@
         <artifactId>maven-compiler-plugin</artifactId>
         <version>3.3</version>
         <configuration>
-          <source>1.7</source>
-          <target>1.7</target>
+          <source>1.8</source>
+          <target>1.8</target>
```
Comment (dorian-burihabwa-sonarsource) [inline its/projects/project-default-config/pom.xml:25]: Should we restore back to 1.7?

File: its/projects/shared/struts-1.3.9-diet/core/pom.xml (hunk 2 above)
Comment (dorian-burihabwa-sonarsource) [inline its/projects/shared/struts-1.3.9-diet/core/pom.xml:28]: It does not look like we are explicitly testing for the version containing commas. Is this change expected here?

**Review by dorian-burihabwa-sonarsource — CHANGES_REQUESTED** [review]

File: its/projects/shared/struts-1.3.9-diet/core/pom.xml (hunk 2 above)
Comment (ADarko22) [reply its/projects/shared/struts-1.3.9-diet/core/pom.xml:28]: I chose to have commas in the version in order to be consistent with the jar having commas since putting commas in the directory/package names would affect all files below.
Anyway, it's probably better to go for the "invoker" test

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
